app.js

```diff
@@ -256,16 +298,53 @@ app.post("/search", (req, res) => {
 	}
 });
 
-// Route for sign up page			  
+	// ======
+	// // Auth Routes
+	// =======
+
+// Show sign up page			  
 app.get("/signup", (req, res) => {
 	var css = ["header", "footer", "global"];
 	res.render("signup", {css: css});
 });				  
-// Route for login page
+// Handle sign up logic
+app.post("/signup", (req, res) => {
+	let newUser = {username: req.body.username, email: req.body.email}
+	User.register(newUser, req.body.password, (err, user) => {
+		if(err) {
+			console.log(err);
+			return res.render("signup");
+		} else {
+			passport.authenticate("local")(req, res, () => {
+				console.log(user);
+				res.redirect("/courses");
+			});
+		}
+	});
+});
+
+
+// Route to show login page
 app.get("/login", (req, res) => {
 	var css = ["header", "footer", "global"];
 	res.render("login", {css: css});
 });	
+
+// Route to handle logic from log in
+app.post("/login", passport.authenticate("local", 
+	{
+		successRedirect: "/courses",
+		failureRedirect: "/login"
+	}), (req, res) => {
+});
+
+// Route to Log Out 
+app.get("/logout", (req, res) => {
+	req.logout();
+	redirect("courses");
```

Comment:
Should be - 
res.redirect("courses");

Reply:
this is correct in my version not sure why there is an error here.